File: tps/entrada e saida/tp01/TP01Q06 - Is em Java/Programa.java
import java.util.*; 

class Programa {

    public static boolean isfim(String b) {
        if(b.charAt(0) == 'F' && b.charAt(1) == 'I' && b.charAt(2) == 'M' && b.length() == 3) 
        return false;
        return true;
        
    }

    public static void is(String texto) {
    texto.toLowerCase();
    System.out.print(isVogal(texto)  + " ");
    System.out.print(isconsu(texto)  + " ");
    System.out.print(isint(texto)  + " ");
    System.out.println(isreal(texto));
    }

    public static String isVogal(String texto) {
     for (int i = 0; i < texto.length(); i++) {
        if (texto.charAt(i) != 'a' && texto.charAt(i) != 'e' && texto.charAt(i) != 'i' && texto.charAt(i) != 'o' && texto.charAt(i) != 'u') {
            return ("NAO");
        }
     }
     return ("SIM");
    }

    public static String isconsu(String texto) {
        for (int i = 0; i < texto.length(); i++) {
           if (!(texto.charAt(i) >= 'a' && texto.charAt(i) <= 'z') || (texto.charAt(i) == 'a' || texto.charAt(i) == 'e' || texto.charAt(i) == 'i' || texto.charAt(i) == 'o' || texto.charAt(i) == 'u')) {
               return ("NAO");
           }
        }
        return ("SIM");
       }

    public static String isint(String texto) {
        for (int i = 0; i < texto.length(); i++) {
           if (!(texto.charAt(i) >= '0' && texto.charAt(i) <= '9')) {
            return ("NAO");
           }
        }
        return ("SIM");
       }
       
       public static String isreal(String texto) {
        int count = 0;
        for (int i = 0; i < texto.length(); i++) {
           if (!(((texto.charAt(i) >= '0' && texto.charAt(i) <= '9') || texto.charAt(i) == ',' || texto.charAt(i) == '.') && count < 2)) {
            return ("NAO");
           }
           if (texto.charAt(i) == ',' || texto.charAt(i) == '.') {
            count++;
           }
        }

        return ("SIM");
       }     

    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] str = new String[1000];
        int y = 0; // numero de strings
        // ler palavras
        do {
            str[y] = sc.nextLine(); 
        } while (isfim(str[y++]));
        y--; //desconsiderar FIM

        for (int i = 0; i < y; i++) {
            is(str[i]);
            }
    }

    
}

File: tps/entrada e saida/tp01/TP01Q07 - Is em C/programa.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include <stdlib.h>
#include <ctype.h>

#define nDeStrings 1000
#define tam 400

bool isfim(char *a);
void is(char *texto);
char *isvogal(char *texto);
char *isconsu(char *texto);
char *isint(char *texto);
char *isreal(char *texto);

int main (void) {

char str[nDeStrings][tam + 1];
int y = 0; // tamanho string
char teste[tam + 1];
    // pegar strings


    do {
        fgets(teste, tam, stdin);
        teste[strcspn(teste, "\r\n")] = 0;
        strcpy(str[y], teste);
    } while (isfim(str[y++]));
    y--; //desnconsiderar fim

      for (int i = 0; i < y; i++) {
        is(str[i]);
    } 

}

bool isfim(char *a) {
    if (a[0] == 'F' && a[1] == 'I' && a[2] == 'M'){
    return false;
    }
    return true;
}

void is(char *texto) {
    for(int i = 0; texto[i]; i++){
  texto[i] = tolower(texto[i]);
}
    printf("%s ", isvogal(texto));
    printf("%s ", isconsu(texto));
    printf("%s ", isint(texto));
    printf("%s\n", isreal(texto));
}

char *isvogal(char *texto) {
    for (int i = 0; i < strlen(texto); i++) {
        if (texto[i] != 'a' && texto[i] != 'e' && texto[i] != 'i' && texto[i] != 'o' && texto[i] != 'u') {
            return ("NAO");
        }
     }
     return ("SIM");
}

char *isconsu(char *texto) {
    for (int i = 0; i < strlen(texto); i++) {
           if (!(texto[i] >= 'a' && texto[i] <= 'z') || (texto[i] == 'a' || texto[i] == 'e' || texto[i] == 'i' || texto[i] == 'o' || texto[i] == 'u')) {
               return ("NAO");
           }
        }
        return ("SIM");
}

char *isint(char *texto) {
     for (int i = 0; i < strlen(texto); i++) {
           if (!(texto[i] >= '0' && texto[i] <= '9')) {
            return ("NAO");
           }
        }
        return ("SIM");
}

char *isreal(char *texto) {
    int count = 0;
        for (int i = 0; i < strlen(texto); i++) {
           if (!(((texto[i] >= '0' && texto[i] <= '9') || texto[i] == ',' || texto[i] == '.') && count < 2)) {
            return ("NAO");
           }
           if (texto[i] == ',' || texto[i] == '.') {
            count++;
           }
        }

        return ("SIM");
}